storage/statedb/cache.go

```diff
@@ -1,8 +1,77 @@
 package statedb
 
+import "errors"
+
+type CacheType string
+
+const (
+	// Available stateDB cache types
+	LocalCache  CacheType = "LocalCache"
+	RemoteCache           = "RemoteCache"
+	HybridCache           = "HybridCache"
+)
+
+var (
+	errNotSupportedCacheType = errors.New("not supported stateDB Cache type")
+)
+
 // Cache interface the cache of stateDB
 type Cache interface {
 	Set(k, v []byte)
 	Get(k []byte) []byte
 	Has(k []byte) ([]byte, bool)
 }
+
+// NewCache creates one type of any supported stateDB caches.
+// TODO-Klaytn: refine input parameters after setting node flags
+func NewCache(cacheType CacheType, maxBytes int, redisEndpoint []string, redisCluster bool) (Cache, error) {
+	switch cacheType {
+	case LocalCache:
+		return NewFastCache(maxBytes), nil
+	case RemoteCache:
+		return NewRedisCache(redisEndpoint, redisCluster)
+	case HybridCache:
+		return NewHybridCache(maxBytes, redisEndpoint, redisCluster)
+	default:
+	}
+	return nil, errNotSupportedCacheType
+}
+
+func NewHybridCache(maxBytes int, endpoint []string, isCluster bool) (Cache, error) {
+	redis, err := NewRedisCache(endpoint, isCluster)
+	if err != nil {
+		return nil, err
+	}
+	return &hybridCache{
+		local:  NewFastCache(maxBytes),
+		remote: redis,
+	}, nil
+}
+
+// hybridCache integrates two kinds of caches: local, remote.
+// local cache uses memory of the local machine and remote cache uses memory of the remote machine.
+type hybridCache struct {
+	local  Cache
+	remote Cache
+}
+
+func (cache *hybridCache) Set(k, v []byte) {
+	cache.local.Set(k, v)
+	cache.remote.Set(k, v)
```

Comment:
Is there a chance that setting values to remote cache takes longer than we expect?
```suggestion
	go cache.remote.Set(k, v)
```

Reply:
It might. However, the suggested change will generate too many goroutines. As you know, more than thousands of items can be set on the stateDB cache in 1 second. I think we better adjust the expected set time with "write timeout" options.